pkg/scale/bitvec.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package scale
+
+const (
+	// maxLen equivalent of `ARCH32BIT_BITSLICE_MAX_BITS` in parity-scale-codec
+	maxLen = 268435455
+	// byteSize is the number of bits in a byte
+	byteSize = 8
+)
+
```

Comment:
I would suggest including this type assertion, we have those in the code to ensure in comptime if a type implements the interface
```suggestion
var _ BitVec = (*bitVec)(nil)
```